packages/autorest.python/autorest/codegen/templates/serialization.py.jinja2

```diff
@@ -105,7 +105,7 @@ class RawDeserializer:
             # Remove Byte Order Mark if present in string
             data_as_str = data_as_str.lstrip(_BOM)
 
-        if content_type is None:
+        if content_type is None or content_type == "text/plain":
```

Comment:
We solved this problem in azure-core already a while back, I would prefer to stay consistent with azure-core:
https://github.com/Azure/azure-sdk-for-python/blob/999ded5c8cdca182359d40265f08ed10f070c548/sdk/core/azure-core/azure/core/pipeline/policies/_universal.py#L643-L644

Reply:
Agree with Laurent that we had better keep consistent with azure-core